src/pyuvdata/utils/uvcalibrate.py

```diff
@@ -402,43 +398,44 @@ def uvcalibrate(
                         raise ValueError(
                             f"Time {this_time} exists on UVData but not on UVCal."
                         )
-                if len(uvcal_times_to_keep) < uvcal.Ntimes:
-                    downselect_cal_times = True
 
-    downselect_cal_freq = False
-    if uvcal.freq_array is not None:
-        uvdata_freq_arr_use = uvdata.freq_array
-        uvcal_freq_arr_use = uvcal.freq_array
-        try:
-            freq_arr_match = np.allclose(
-                np.sort(uvcal_freq_arr_use),
-                np.sort(uvdata_freq_arr_use),
-                atol=uvdata._freq_array.tols[1],
-                rtol=uvdata._freq_array.tols[0],
-            )
-        except ValueError:
-            freq_arr_match = False
-
-        if freq_arr_match is False:
-            # check more carefully
-            uvcal_freqs_to_keep = []
-            for this_freq in uvdata_freq_arr_use:
-                wh_freq_match = np.nonzero(
-                    np.isclose(
-                        uvcal.freq_array - this_freq,
-                        0,
+    uvcal_chans_to_keep = None if uvcal.wide_band else []
+    uvcal_spws_to_keep = [] if uvcal.wide_band else None
+    for idx, spw in enumerate(uvcal.spw_array):
+        if spw in uvdata.spw_array:
+            if uvcal.wide_band:
+                freq_array_spw = uvdata.freq_array[uvdata.flex_spw_id_array == spw]
+                min_freq = min(uvcal.freq_range[idx])
+                max_freq = max(uvcal.freq_range[idx])
+                if all((freq_array_spw < min_freq) | (freq_array_spw > max_freq)):
+                    raise ValueError(
+                        "SPWs between UVData and UVCal overlap, but the frequencies of "
+                        "channels are inconsistent with frequency ranges."
+                    )
+                uvcal_spws_to_keep.append(spw)
```

Comment:
It seems like either all the spws end up in this list or it errors, so why bother building it up this way? Why not just set it equal to uvdata.spw_array after doing this check per spw?

Reply:
The original logic I had set up wasn't _quite_ right -- it now better mimics what's present for the frequency axis in non-wideband gains.